Objects/rangeobject.c

```diff
@@ -572,16 +572,62 @@ range_count(rangeobject *r, PyObject *ob)
 }
 
 static PyObject *
-range_index(rangeobject *r, PyObject *ob)
+range_index(rangeobject *r, PyObject *args, PyObject *kw)
 {
     int contains;
+    static char *keywords[] = {"value", "start", "stop", NULL};
+    PyObject *ob;
+    PyObject *start = NULL, *stop = NULL;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kw, "O|OO:range_index",
```

Comment:
I think that it is worth to use Argument Clinic. Clearer code and less overhead.